client/src/modules/list/components/ListItem.jsx

```diff
@@ -41,16 +60,118 @@ class ListItem extends PureComponent {
     // Adding new comment will be handled in next tasks
   };
 
+  handleDataUpdate = () => {
+    const { itemDescription, link } = this.state;
+    const {
+      data: { description: previousDescription, link: previousLink }
+    } = this.props;
+    const isLinkUpdated = !areStringsEqual(previousLink, link);
+    const isDescriptionUpdated = !areStringsEqual(
+      previousDescription,
+      itemDescription
+    );
+
+    if (isLinkUpdated) {
+      this.handleLinkUpdate();
+    }
+
+    if (isDescriptionUpdated) {
+      this.handleDescriptionUpdate();
+    }
+  };
+
+  handleLinkUpdate = () => {
+    const { link } = this.state;
+    const {
+      addItemLink,
+      data: { _id: itemId },
+      match: {
+        params: { id: listId }
+      }
+    } = this.props;
+    const canBeUpdated = isUrlValid(link) || _isEmpty(_trim(link));
+
+    if (canBeUpdated) {
+      addItemLink(listId, itemId, link);
+    } else if (!isUrlValid(link)) {
+      this.setState({ isValidationErrorVisible: true });
+    }
+  };
+
+  handleDescriptionUpdate = () => {
+    const { itemDescription } = this.state;
+    const {
+      addItemDescription,
+      data: { _id: itemId },
+      match: {
+        params: { id: listId }
+      }
+    } = this.props;
+
+    addItemDescription(listId, itemId, itemDescription);
```

Comment:
on one click save button, we may have to API calls at the same time to update one item: addItemDescription, addItemLink. I Wonder if it could be done with one request like in creating a list/cohort case? 
What if both failed will we receive two notifications?